### ISSUE TITLE
PlatformEventSubscriberConfig to avoid need for Org-wide Email?

### ISSUE THREAD

**Issue by kacrouse** [j-fischer/rflib#70]
**Is your feature request related to a problem? Please describe.**
It's annoying to have to configure an org-wide email address for alert emails to be able to be sent out. Also not ideal that the email has to be available to all profiles.

**Describe the solution you'd like**
The PlatformEventSubscriberConfig might be a way to avoid the need for an org-wide email. I haven't tested it so I could be wrong... 

I also imagine it would come with some setup on the admin's part since it is tied to a user in the org, has no user interface for configuration, and I'm unsure if it is packageable.

**Additional context**
Some good info on PlatformEventSubscriberConfig https://salesforce.stackexchange.com/questions/357570/how-do-i-change-the-running-user-of-a-platform-event-trigger-handler

**Comment by j-fischer** [comment]
Hi @kacrouse ,

Thanks for opening this ticket. Unfortunately, I don't believe that what you are asking for can be done. `PlatformEventSubscriberConfig` only work with triggers. However, RFLIB handles the platform event with the Platform Event Triggered Flow, so I cannot configure the running user for the handler. Also, I would not be able to package it as the config requires a valid user name and I would not be able to provide this for users. So, if anything is possible, it would have to be an optional setting. 

Since the Email Action runs with a system user from a triggered platform event flow, you have to provide an Org Wide Email Address when sending the message. I am not sure what else can be done here but I am open for suggestions.

Please reopen the issue if you have any other ideas. I am open for suggestions. 

Thanks,
Johannes

PS: Sorry for the delayed responses. I was busy at work for the last 3 weeks.

**Comment by kacrouse** [comment]
Thanks for the response! I agree with the limitations you're calling out. It would be nice to have as an optional setting, but it does look like that would require some significant rework given the current setup of using Flow, and may not be worth the effort. Maybe they will allow PlatformEventSubscriberConfig to apply to Flows eventually (🤞) at which point it probably makes more sense.

**Comment by kacrouse** [comment]
Spring '24 might be giving us a solution to this one! 

[Run Event-Triggered Flows as Workflow User](https://help.salesforce.com/s/articleView?id=release-notes.rn_automate_flow_builder_run_event_triggered_flows_as_workflow_user.htm&release=248&type=5)

**Comment by j-fischer** [comment]
Hi @kacrouse , thanks for posting this. It sounds interesting. I am happy to explore this over the break in a preview org. Will keep you posted, but please don't hesitate to provide more details on how you would like this to be done. 

Thanks

**Comment by j-fischer** [comment]
@kacrouse I tested your suggestion in the Preview scratch org and everything is looking pretty good. 
I needed to use a backwards compatible approach, which uses a CMT record to identify whether or not to use the an org wide email address (see linked PR #81) , which means that an upgrade can reset your Global Settings values. Just something to be aware of. Have a look and let me know what you think. 

Planning to release this version shortly after Spring 24 is out. 

Cheers
